### DISCUSSION TITLE
Rename folder in bucket

### DISCUSSION THREAD

**Discussion by AlexZIX** [minio/minio#18064] [Q&A] [answered]
Hi,

is it possible to rename folder (key) inside bucket? Or I should download object and then upload it with the new key name?

**Comment by klauspost** [comment, marked as answer]
No. You can copy objects with [CopyObject](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CopyObject.html) API to avoid download+upload.

**Reply by mshahulpm** [reply]
Is this api available in nodejs or golang sdk ?

**Reply by klauspost** [reply]
Yes.
